src/components/ComputeModal.js

```diff
@@ -0,0 +1,1335 @@
+}
+
+const getCurrentPersistentDisk = (runtimes, persistentDisks) => {
+  const currentRuntime = getCurrentRuntime(runtimes)
+  const id = currentRuntime?.runtimeConfig.persistentDiskId
+  const attachedIds = _.without([undefined], _.map(runtime => runtime.runtimeConfig.persistentDiskId, runtimes))
+
+  return id ?
+    _.find({ id }, persistentDisks) :
+    _.last(_.sortBy('auditInfo.createdDate', _.filter(({ id, status }) => status !== 'Deleting' && !_.includes(id, attachedIds), persistentDisks)))
+}
+
+const shouldUsePersistentDisk = (sparkMode, runtimeDetails, upgradeDiskSelected) => !sparkMode &&
+  (!runtimeDetails?.runtimeConfig?.diskSize || upgradeDiskSelected)
+
+export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDisks, tool, workspace, isAnalysisMode = false }) => {
+  // State -- begin
+  const [showDebugger, setShowDebugger] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const [currentRuntimeDetails, setCurrentRuntimeDetails] = useState(() => getCurrentRuntime(runtimes))
+  const [currentPersistentDiskDetails, setCurrentPersistentDiskDetails] = useState(() => getCurrentPersistentDisk(runtimes, persistentDisks))
+  const [viewMode, setViewMode] = useState(undefined)
+  const [deleteDiskSelected, setDeleteDiskSelected] = useState(false)
+  const [upgradeDiskSelected, setUpgradeDiskSelected] = useState(false)
+  const [simplifiedForm, setSimplifiedForm] = useState(!currentRuntimeDetails)
+  const [leoImages, setLeoImages] = useState([])
+  const [selectedLeoImage, setSelectedLeoImage] = useState(undefined)
+  const [customEnvImage, setCustomEnvImage] = useState(undefined)
+  const [jupyterUserScriptUri, setJupyterUserScriptUri] = useState(undefined)
+  const [sparkMode, setSparkMode] = useState(false)
+  const [computeConfig, setComputeConfig] = useState({
+    selectedPersistentDiskSize: defaultGcePersistentDiskSize,
+    masterMachineType: defaultGceMachineType,
+    masterDiskSize: defaultGceBootDiskSize,
+    numberOfWorkers: defaultNumDataprocWorkers,
+    numberOfPreemptibleWorkers: defaultNumDataprocPreemptibleWorkers,
+    workerMachineType: defaultDataprocMachineType,
+    workerDiskSize: defaultDataprocDiskSize,
+    gpuEnabled: false,
+    hasGpu: false,
+    gpuType: defaultGpuType,
+    numGpus: defaultNumGpus
+  })
+  // State -- end
+
+  const isPersistentDisk = shouldUsePersistentDisk(sparkMode, currentRuntimeDetails, upgradeDiskSelected)
+
+  const isCustomImage = selectedLeoImage === customMode
+  const { version, updated, packages, requiresSpark, label: packageLabel } = _.find({ image: selectedLeoImage }, leoImages) || {}
+
+  const minRequiredMemory = sparkMode ? 7.5 : 3.75
+  const validMachineTypes = _.filter(({ memory }) => memory >= minRequiredMemory, machineTypes)
+  const mainMachineType = _.find({ name: computeConfig?.masterMachineType }, validMachineTypes)?.name || getDefaultMachineType(sparkMode)
```

Comment:
`computeConfig` state variable should be defined but opted to use the optional chaining operator (`?`) when referencing a property of its in case I was missing something.